### PR TITLE
Improve documentation on building Flatcar

### PR DESCRIPTION
**Pull request by pothos**
The existing docs were too brief and started with some confusing advice
to set up a password which is not really needed in most cases.
Give more context on how things work and instruct on preparing the
correct build branches and binary package references.

### CODE REVIEW

File: os/sdk-modifying-flatcar.md [hunk 1]
```diff
@@ -87,14 +94,52 @@ To use the SDK chroot in the future, run `cork enter` from the above directory.
 
 #### Set up the chroot
 
-After entering the chroot via `cork` for the first time, you should set user `core`'s password:
+When entering the SDK you are in the `scripts` repository whih can be seen as the build system.
+It is one of the three repositories that define a Flatcar build. The other two are the `../third_party/portage-stable`
```
Comment (dongsupark) [inline os/sdk-modifying-flatcar.md:98]: I would rather say like `~/trunk/src/third_party/portage-stable`, as `..` varies on where you are right now.

File: os/sdk-modifying-flatcar.md [hunk 2]
```diff
@@ -87,14 +94,52 @@ To use the SDK chroot in the future, run `cork enter` from the above directory.
...
+and the `../third_party/coreos-overlay` repository. These contain Gentoo package build files. The `portage-stable` repository
```
Comment (dongsupark) [inline os/sdk-modifying-flatcar.md:99]: Ditto

File: os/sdk-modifying-flatcar.md [hunk 3]
```diff
@@ -87,14 +94,52 @@ To use the SDK chroot in the future, run `cork enter` from the above directory.
...
+hosts build files for upstream Gentoo packages and the `coreos-overlay` repository contains additional and patched packages.
+You have to check out the right combination of branches if you don't use the default. For example, checking out a feature branch
+based on the `flatcar-master-alpha` branch of `coreos-overlay` means that `portage-stable` and `scripts` also need to point
+to `flatcar-master-alpha` (or a feature branch based on it).
+Any other repositories under `../third_party/` are managed by commit references and you should not alter them.
```
Comment (dongsupark) [inline os/sdk-modifying-flatcar.md:104]: Ditto

File: os/sdk-modifying-flatcar.md [hunk 4]
```diff
@@ -87,14 +94,52 @@ To use the SDK chroot in the future, run `cork enter` from the above directory.
 
 #### Set up the chroot
 
-After entering the chroot via `cork` for the first time, you should set user `core`'s password:
+When entering the SDK you are in the `scripts` repository whih can be seen as the build system.
```
Comment (dongsupark) [inline os/sdk-modifying-flatcar.md:97]: I would rather explicitly say that you are in the `~/trunk/src/scripts` directory.
Then all the following mentions about the `scripts` directory as well as `./some-script` could be clearer.

Comment (sayanchowdhury) [inline os/sdk-modifying-flatcar.md:97]: s/whih/which/

File: os/sdk-modifying-flatcar.md [hunk 5]
```diff
@@ -87,14 +94,52 @@ To use the SDK chroot in the future, run `cork enter` from the above directory.
...
+to `flatcar-master-alpha` (or a feature branch based on it).
+Any other repositories under `../third_party/` are managed by commit references and you should not alter them.
+
+The chroot you entered has the SDK packages installed. The actual image packages are built in a new chroot under `/build/amd64-usr/`
+or `/build/arm64-usr/`, called the _board_.
+Both chroots use Gentoo's portage to manage the packages (via `sudo emerge` for the SDK and `emerge-BOARD` for the boards).
+To reduce the build time it is important that as many binary packages as possible are downloaded from a previous SDK release,
+Flatcar release, or nightly build.
+
+To reference today's nightly builds as source for board packages, use the following command. It has to be rerun if you want to
```
Comment (marga-kinvolk) [inline os/sdk-modifying-flatcar.md:112]: I'd say "the latest nightly" build instead of today's.

File: os/sdk-modifying-flatcar.md (hunk 4 above)
Comment (pothos) [reply os/sdk-modifying-flatcar.md:97]: Thanks, will change!

File: os/sdk-modifying-flatcar.md (hunk 5 above)
Comment (pothos) [reply os/sdk-modifying-flatcar.md:112]: It is really just dereferenced on that day which I tried to make clear. I took your suggestion but added more information in the following sentence.

**Pull request merged**